cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -132,6 +126,20 @@ public void setAuditPublisher(AuditPublisher auditPublisher) {
    * @throws IOException if there is a problem in making an HTTP request to check if the namespace exists.
    */
   public DatasetMeta get(Id.DatasetInstance instance, List<? extends Id> owners) throws Exception {
+    try {
+      return metaCache.get(instance);
+    } catch (ExecutionException e) {
+      if ((e.getCause() instanceof Exception) && (e.getCause() instanceof HttpErrorStatusProvider)) {
```

Comment:
yes we could assume here that it is an exception an cast it. I think it is cleaner though, to check for Exception, because e.getCause() is of type Throwable. If any of the code in the execute changes in the future and throws an error then casting to Exception would break.